### PR TITLE
specification_for_admin_page

### PR DESCRIPTION
**Pull request by V-la-d**

### CODE REVIEW

File: dao/pom.xml [hunk 1]
```diff
@@ -31,9 +31,15 @@
             <groupId>org.projectlombok</groupId>
             <artifactId>lombok</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-jpamodelgen</artifactId>
+            <version>5.4.3.Final</version>
+        </dependency>
```
Comment (forestik) [inline dao/pom.xml:38]: Please, move version to properties section

File: service-api/src/main/java/greencity/filters/SearchCriteria.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        } else {
+            valueEquals = this.value.equals(that.value);
+        }
+        return valueEquals && key.equals(that.key) && type.equals(that.type);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(value, key, type);
+    }
```
Comment (forestik) [inline service-api/src/main/java/greencity/filters/SearchCriteria.java:41]: Above you use @EqualsAndHashCode. Why do you redefine yourself here?

File: service/src/main/java/greencity/service/UserServiceImpl.java [hunk 3]
```diff
@@ -10,22 +12,7 @@
...
-import greencity.dto.user.UserProfileDtoResponse;
-import greencity.dto.user.UserProfilePictureDto;
-import greencity.dto.user.UserProfileStatisticsDto;
-import greencity.dto.user.UserRoleDto;
-import greencity.dto.user.UserStatusDto;
-import greencity.dto.user.UserUpdateDto;
-import greencity.dto.user.UserVO;
-import greencity.dto.user.UserWithOnlineStatusDto;
-import greencity.dto.user.UsersFriendDto;
+import greencity.dto.user.*;
```
Comment (forestik) [inline service/src/main/java/greencity/service/UserServiceImpl.java:15]: Customize your IDEA not to grab imports to *

File: service-api/src/main/java/greencity/filters/SearchCriteria.java (hunk 2 above)
Comment (V-la-d) [reply service-api/src/main/java/greencity/filters/SearchCriteria.java:41]: Because it custom equals and hashCode methods.

Comment (forestik) [reply service-api/src/main/java/greencity/filters/SearchCriteria.java:41]: @EqualsAndHashCode will do the same

Comment (forestik) [reply service-api/src/main/java/greencity/filters/SearchCriteria.java:41]: @EqualsAndHashCode will do the same

File: dao/pom.xml (hunk 1 above)
Comment (V-la-d) [reply dao/pom.xml:38]: done

File: service/src/main/java/greencity/service/UserServiceImpl.java (hunk 3 above)
Comment (V-la-d) [reply service/src/main/java/greencity/service/UserServiceImpl.java:15]: done

File: service-api/src/main/java/greencity/filters/SearchCriteria.java (hunk 2 above)
Comment (V-la-d) [reply service-api/src/main/java/greencity/filters/SearchCriteria.java:41]: fix

**Review by forestik — APPROVED** [review]

**Review by Skajl-dev — APPROVED** [review]

**Review by MarianDiakiv — APPROVED** [review]

**Pull request merged**